### DISCUSSION TITLE
API difficulties on containers/stats

### DISCUSSION THREAD

**Discussion by squashlala** [containers/podman#23459] [Q&A] [answered]
Hello, I'd like to get the stats of my running containers but I'm stuck on the API call on /libpod/containers/stats through unix socket (cf. https://docs.podman.io/en/latest/_static/api.html?version=v5.0#tag/containers/operation/ContainersStatsAllLibpod)

Here is the curl command I'm using :
curl -N --silent --unix-socket /run/user/0/podman/podman.sock http://d/containers/stats

But dispite my efforts, I'm unable to get any result beside a "405 : Method Not Allowed" as you can see below.
![image](https://github.com/user-attachments/assets/652d0cdc-ddbf-4f6b-88a6-00a5668f3fc3)

Could someone help me figure it out please?

**Comment by Luap99** [comment, marked as answer]
Your screenshot shows /containers/stats while link you posted says /libpod/containers/stats, you also have to include the version first so `/v5.0.0/libpod/containers/stats`

**Reply by squashlala** [reply]
Thank very much!
I've went head on with the `/containers/stats` that I haven't even tried this way...

May I ask you why some GETs like `/containers/json` are working this way when other such as the stats requires the full `/<version>/libpod/<...>`  path? Is it just due to a redesign of the API?

**Reply by Luap99** [reply]
docker allows API requests with and without a version set so all the normal calls are used by the compat API layer.
For podman/libpod API we needed a special prefix such as /libpod to not cause conflicts with docker endpoints and we decided to always require a version (I am not sure why)
